benchmarks/cpp/transformer.cpp

```diff
@@ -0,0 +1,194 @@
+// clang-format off
+/*
+ * SPDX-FileCopyrightText: Copyright (c) 2023-present NVIDIA CORPORATION & AFFILIATES.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+// clang-format on
+#include <device_lower/lower2device.h>
+#include <fusion.h>
+#include <ir/all_nodes.h>
+#include <ir/builder.h>
+#include <ir/utils.h>
+#include <ops/all_ops.h>
+#include <runtime/executor.h>
+#include <scheduler/all_schedulers.h>
+#include <scheduler/utils.h>
+
+#include <benchmark/benchmark.h>
+
+#include <cuda_runtime.h>
+
+#include <sstream>
+
+#include <benchmarks/cpp/utils.h>
+#include <csrc/multidevice/utils.h>
+#include <tests/cpp/utils.h>
+#include <tests/cpp/multidevice_transformer.h>
+
+using namespace nvfuser;
+
+namespace {
+// Note: We test on smaller model and input sizes to avoid high error
+// accumulation for validation.
+static constexpr int64_t B = 2, E = 768, H = 16, S = 128;
+// Note: Dropout probabilities are set to 0. Since the dropout mask is sharded
+// it throws off the seed offset between the sharded nvFuser program and the
+// unsharded reference.
+static constexpr double kDropoutProb = 0.0, kSdpaProb = 0.0, kSdpaScale = 1e-3;
+// Note parameters scaled by kParamScale following weight initialization
+// recommendations:
+// https://huggingface.co/docs/transformers/en/model_doc/gpt2#transformers.GPT2Config.initializer_range
+static constexpr double kParamScale = 0.02;
+} // namespace
+
+// Return reduction tensor view and output of reduction
+static void setupTransformerForward(Fusion* fusion, DataType dtype) {
+  Communicator* communicator_ = &Communicator::getInstance(); // nick TODO call Communicator::getInstance().cleanup() somewhere before program exit
+
+  const int64_t D = communicator_->size(); // number of devices
+
+  std::unique_ptr<DistributedTransformer> model = std::make_unique<DistributedTransformer>(
+        D, B, E, H, S, kDropoutProb, kSdpaProb);
+
+  model->setupForward(fusion, dtype, /*sequence_parallel*/false);
```

Comment:
```suggestion
  model->setupForward(fusion, dtype, /*sequence_parallel=*/false);
```
https://clang.llvm.org/extra/clang-tidy/checks/bugprone/argument-comment.html